DESIGN.md

```diff
@@ -0,0 +1,36 @@
+### Define Your Schema with Zod Scheming
+
+MarkdownDB leverages the powerful Zod library for schema definition, allowing you to specify the structure and validation rules for your Markdown content. Below is an example configuration demonstrating the usage of Zod to define a schema for a blog post with a mandatory `date` field.
+
+```javascript
+// markdowndb.config.js
+import { z } from "zod";
+
+module.exports = {
+  schemas: z.object({
+    post: {
+      date: z.string().refine((value) => /\d{4}-\d{2}-\d{2}/.test(value), {
+        message: "Invalid date format. Please use YYYY-MM-DD.",
+      }),
+      // Add more fields as needed, each with its own validation rules
+      // Example:
+      // title: z.string().min(1, "Title must have at least 1 character"),
+      // content: z.string(),
+    },
+    // Define additional schemas for different content types
+    // Example:
+    // page: {
+    //   author: z.string(),
+    //   text: z.string(),
+    // },
+  }),
+};
+```
+
+In this example, the `post` schema is defined using Zod's `object` method, specifying the structure of the content. The `date` field is required and must adhere to the specified date format using Zod's `string` and `refine` methods for custom validation.
+
+### How Validation Works
```

Comment:
Could you provide a short example of what output will look like?

Reply:
@rufuspollock 

So I think what you mean by `output` is either:
1- Error messages if a given file e.g. `blog.md` is missing a field e.g. `author` will be `Error: Missing 'author' field in 'blog.md' for the 'POST' scheme. Please ensure that the required field is included in the file.`

2- Typescript declaration files `.d.ts` for nextjs Typescript users.

Reply:
OK. That's fine.